File: example-java/app/src/main/java/com/vibes/androidsdkexampleapp/activities/MainActivity.java
package com.vibes.androidsdkexampleapp.activities;

import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.vibes.androidsdkexampleapp.R;
import com.vibes.androidsdkexampleapp.fragments.InboxMessagesFragment;
import com.vibes.androidsdkexampleapp.fragments.VibesMainFragment;

/**
 * Created by jean-michel.barbieri on 2/25/18
 * Copyright (c) Vibes 2018 . All rights reserved.
 * Last modified 12:33 AM
 */
public class MainActivity extends AppCompatActivity {
    public static final String INBOX_MESSAGE_KEY = "message";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
        loadFragment(new VibesMainFragment());
        bottomNavigationView.setOnNavigationItemSelectedListener(item -> {
            Fragment fragment;
            switch (item.getItemId()) {
                case R.id.action_home:
                    item.setEnabled(true);
                    fragment = new VibesMainFragment();
                    loadFragment(fragment);
                    return true;
                case R.id.action_inbox_messages:
                    item.setEnabled(true);
                    fragment = new InboxMessagesFragment();
                    loadFragment(fragment);
                    return true;
            }
            return false;
        });
    }

    private void loadFragment(Fragment fragment) {
        String backStateName = fragment.getClass().getName();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.frame_container, fragment);
        transaction.addToBackStack(backStateName);
        transaction.commit();
    }

    @Override
    public void onBackPressed() {
        if (getFragmentManager().getBackStackEntryCount() > 0) {
            getFragmentManager().popBackStack();
        } else {
            super.onBackPressed();
        }
    }
}

File: example-kotlin/app/src/main/java/com/android/exampleapp/kotlin/MainActivity.kt
package com.android.exampleapp.kotlin

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.android.exampleapp.kotlin.fragments.InboxMessagesFragment
import com.android.exampleapp.kotlin.fragments.VibesMainFragment
import com.google.android.material.bottomnavigation.BottomNavigationView


class MainActivity : AppCompatActivity() {

    companion object {
        val INBOX_MESSAGE_KEY = "message"
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        loadFragment(VibesMainFragment())
        bottomNavigationView.setOnNavigationItemSelectedListener { item: MenuItem ->
            val fragment: androidx.fragment.app.Fragment
            when (item.itemId) {
                R.id.action_home -> {
                    item.isEnabled = true
                    fragment = VibesMainFragment()
                    loadFragment(fragment)
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.action_inbox_messages -> {
                    item.isEnabled = true
                    fragment = InboxMessagesFragment()
                    loadFragment(fragment)
                    return@setOnNavigationItemSelectedListener true
                }
            }
            false
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            val permissionState =
                ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS)

            // If the permission is not granted, request it.
            if (permissionState == PackageManager.PERMISSION_DENIED) {
                ActivityCompat.requestPermissions(
                    this,
                    arrayOf(Manifest.permission.POST_NOTIFICATIONS),
                    1
                )
            }
        }
    }

    private fun loadFragment(fragment: androidx.fragment.app.Fragment) {
        val backStateName = fragment.javaClass.name
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.frame_container, fragment)
        transaction.addToBackStack(backStateName)
        transaction.commit()
    }

    override fun onBackPressed() {
        if (fragmentManager.backStackEntryCount > 0) {
            fragmentManager.popBackStack()
        } else {
            super.onBackPressed()
        }
    }
}
